2022Swerve/src/main/java/frc/robot/subsystems/AutonSubsytem.java

```diff
@@ -0,0 +1,137 @@
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.util.datalog.StringLogEntry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Filesystem;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.SwerveControllerCommand;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import frc.robot.Constants.AutonConstants;
+import frc.robot.Constants.DriveConstants;
+
+
+public class AutonSubsytem extends SubsystemBase{
+    private enum AutonModes{
+        FORWARD, // Go forward 2 meters
+        BACKWARD, // Wait 3 seconds, go backward 2 meters
+        FORWARD180; // Go forward 2 meters and rotate 180 degrees
+    }
+
+    private ShuffleboardTab configTab = Shuffleboard.getTab("Config");
+    private NetworkTableEntry delayEntry = configTab.add("Auton Delay", 0.0).getEntry();
+    private SendableChooser<AutonModes> modeChooser = new SendableChooser<>();
+
+    private DataLog datalog = DataLogManager.getLog();
+    private StringLogEntry trajectoryLog = new StringLogEntry(datalog, "/auton/trajectory"); //Logs x translation state output
+
+    private SwerveSubsystem swerveSubsystem;
+
+    private PIDController xController;
+    private PIDController yController;
+    private ProfiledPIDController rotationController;
+
+    private AutonModes autonMode;
+
+    public AutonSubsytem(SwerveSubsystem swerveSubsystem){
+        this.swerveSubsystem = swerveSubsystem;
+
+        //Add auton modes to chooser
+        for(AutonModes mode : AutonModes.values()){
+            modeChooser.addOption(mode.toString(), mode);
+        }
+        configTab.add("Auton mode", modeChooser);
+
+        //Define PID controllers for tracking trajectory
+        xController = new PIDController(AutonConstants.kXTranslationP, 0, 0);
+        yController = new PIDController(AutonConstants.kYTranslationP, 0, 0);
+        rotationController = new ProfiledPIDController(
+            AutonConstants.kRotationP, 0, 0, AutonConstants.kRotationalConstraints);
+        rotationController.enableContinuousInput(-Math.PI, Math.PI);
+    }
+
+    private Trajectory getTrajectory(String name){
+        Trajectory trajectory;
+        try {
+            Path trajectoryPath = Filesystem.getDeployDirectory().toPath().resolve("pathplanner/generatedJSON/" + name + ".wplib.json");
+            trajectory = TrajectoryUtil.fromPathweaverJson(trajectoryPath);
+        } catch (IOException ex) {
+            DriverStation.reportError("Unable to open trajectory: " + name, ex.getStackTrace());
+            trajectory = null;
```

Comment:
Can we check and see if this also goes to high speed logging?  If not, let's also log there.

Thinking through this, is there a default path that we can do in this case?  To get the hanger points no matter what?

Maybe not...and maybe not here...but something to consider.

Reply:
We can do a default path easily, just change the default case in getAutonSequence() 
An error reported to DriverStation should be in the logs, as we set logs to capture all DriverStation data

Reply:
to be 100% clear, I mean something in code, and not depending on reading from a file at all.  Maybe it's not worth it until it becomes a problem

Reply:
I could write a method to generate a trajectory from a pose. However, if for any reason we are using runtime generated trajectories instead of pathplanner trajectories, that is a failure in my opinion, and should only be used as a fallback.

Reply:
Absolutely, it is a terrible failure, and fallback case.

https://trello.com/c/OLmMwKac Card in backlog, resolving